src/many-abci/src/module.rs

```diff
@@ -102,6 +119,79 @@ fn _tm_query_from_many_filter(
     Ok(query)
 }
 
+fn transform_list_args(
+    blockchain::ListArgs {
+        count,
+        order,
+        filter,
+    }: blockchain::ListArgs,
+) -> Result<(u64, Order, Query), ManyError> {
+    filter
+        .map_or(
+            Ok(DEFAULT_BLOCK_LIST_QUERY.clone()),
+            _tm_query_from_many_filter,
+        )
+        .map(|filter| {
+            (
+                count.map_or(MAXIMUM_BLOCK_COUNT, |c| {
+                    std::cmp::min(c, MAXIMUM_BLOCK_COUNT)
+                }),
+                order.map_or(tendermint_rpc::Order::Ascending, _tm_order_from_many_order),
+                filter,
+            )
+        })
+}
+
+fn create_pagination(count: u64) -> (u64, u64) {
+    // We can get maximum u8::MAX blocks per page and a maximum of u32::MAX pages
+    // Find the correct number of pages and count
+    let maximum_8_bit_integer: u64 = u8::MAX.into();
+    (
+        num_integer::div_ceil(count, maximum_8_bit_integer),
+        std::cmp::max(count, maximum_8_bit_integer),
+    )
+}
+
+fn tx_results<C: Client + Sync>(
+    client: &C,
+    count: u64,
+    order: Order,
+    query: Query,
+) -> Result<Vec<(TransactionIdentifier, Vec<u8>)>, ManyError> {
+    use futures_util::future::TryFutureExt;
+    let (pages, count) = create_pagination(count);
+    block_on(
+        async {
+            let pages: u32 = pages
+                .try_into()
+                .map_err(|_| ManyError::unknown("Unable to cast u64 to u32"))?;
+            let count: u8 = count
+                .try_into()
+                .map_err(|_| ManyError::unknown("Unable to cast u64 to u8"))?;
```

Comment:
This code block is present twice. Make `create_pagination` return `(u32, u8)` instead.

Reply:
Good catch . . . will take this suggestion